### PR TITLE
Use vscode log path

### PR DESCRIPTION
**Pull request by anthonydresser**
path changes from 
 {temp}\azuredatastudio\mssql\
to 
{temp}\azuredatastudio\logs\{dateStamp}\exthost1\Microsoft.mssql

### CODE REVIEW

File: extensions/mssql/src/credentialstore/credentialstore.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ export class CredentialStore {
 	private _client: SqlOpsDataClient;
 	private _config: IConfig;
 
-	constructor(baseConfig: IConfig) {
+	constructor(private logPath: string, baseConfig: IConfig) {
```
Comment (ranasaria) [inline extensions/mssql/src/credentialstore/credentialstore.ts]: nit: should this be called _logPath?

File: extensions/mssql/src/utils.ts [hunk 2]
```diff
@@ -95,24 +107,20 @@ export function getConfigTracingLevel(): string {
...
+	return `${prefix}_${pid}.log`;
 }
 
-export function getCommonLaunchArgsAndCleanupOldLogFiles(prefix: string, executablePath: string): string[] {
+export function getCommonLaunchArgsAndCleanupOldLogFiles(logPath: string, prefix: string, executablePath: string): string[] {
 	let launchArgs = [];
 	launchArgs.push('--log-file');
-	let logFile = getDefaultLogFile(prefix, process.pid);
+	let logFile = path.join(logPath, getLogFileName(prefix, process.pid));
 	launchArgs.push(logFile);
```
Comment (Charles-Gagnon) [inline extensions/mssql/src/utils.ts]: Oh hey - we're not quoting this path which means it'll break for anyone with a space in the path. I'll make a separate issue for it but if you feel like fixing it now that'd be great =D

**Comment by ranasaria** [comment]
Can you verify that the following issues are addressed:

https://github.com/microsoft/azuredatastudio/issues/4509
https://github.com/microsoft/azuredatastudio/issues/4510

if not we should take a quick look as to why not.

**Comment by anthonydresser** [comment]
It would fix #4509 since thats what the "exthost1" part of the new part handles. I'm not sure about the second one.

**Comment by anthonydresser** [comment]
@ranasaria I think this might be the chance to move the sqltoolslog output from being built into the product to contributed by the extension. Was the goal of that edit to map the sqltools log file to a vscode output window? That should be doable from the extension side using the same lib that vscode uses `sqdlog`

**Comment by ranasaria** [comment]
Yeah they were both related.

Thanks! – I will check out the 2nd one after your fix is in. Can you please reference the 4509 but in your pull request.


-  Arvind.

________________________________

**Comment by ranasaria** [comment]
Yes, I agree. I will sync up with you later to do the needful next week.


-  Arvind.

________________________________

**Comment by anthonydresser** [comment]
I think we will need that change for this PR, since currently this PR will just break the output window, since the log path is now generated, vs in the code you point to a hard coded log path. I can just implement it as part of this PR if you have thoughts on implementation.

**Comment by anthonydresser** [comment]
@ranasaria is it fine if we remove the vscode edit where we add the sqltoolsservice log and instead add an action to the command pallete to open the log file? I think it achieves the same purpose of easily seeing the sqltoolsservice log. Currently there is no built in way to link a file to a output channel from an extension.

**Comment by ranasaria** [comment]
Let's discus the alternatives face to face

Arvind

________________________________

File: extensions/mssql/src/credentialstore/credentialstore.ts (hunk 1 above)
Comment (ranasaria) [reply extensions/mssql/src/credentialstore/credentialstore.ts]: if you do not waht to change to _logPath then consider renaming existing private fields _client and _config to not start with _ for consistency.

**Pull request merged**